### PR TITLE
ENH Create transfer learning tutorial

### PR DESCRIPTION
**Pull request by javadbayazi**
## Summary

Add an executable transfer-learning tutorial that demonstrates how to transfer a pathology classifier from TUAB-style source data to NMT-style target data.

The gallery example is deliberately small, deterministic, CPU-only, and network-free. It uses synthetic recordings so documentation builds remain reliable; the text separately shows the exact public dataset constructors and paths needed for a real experiment.

## What the tutorial demonstrates

- split recordings before creating windows, with stable participant identities and explicit cross-split leakage checks;
- fit source, target-from-scratch, and target-transfer classifiers under the same protocol;
- select models using training/validation data only and defer target-test window and metadata access until all fitting is complete;
- load the source model state strictly after target-model initialization, then fine-tune the complete network;
- reset seeds between comparisons and run on CPU for reproducible gallery output;
- report the tiny synthetic scores as illustrative output, not as a scientific reproduction.

The real-data section pins TUAB `v3.0.1`, identifies the expected dataset parent directory, links the TUAB and NMT sources, and summarizes the preprocessing choices that a full experiment must make.

## Validation

- focused tutorial regressions: 5 passed;
- adjacent dataset, model, normalization, and windowing tests: 25 passed;
- the tutorial ran twice with byte-identical output;
- the target-test access oracle confirms that all three fits finish before target-test windows or metadata are read;
- strict state-transfer and empty pre-step optimizer-state checks pass;
- Sphinx-Gallery generation and a warnings-as-errors targeted Sphinx build pass;
- all applicable pre-commit hooks pass.

The complete candidate was also independently reviewed at exact commit `cfdfa182188dedec2a521f51666931ef2225a087`.

### CODE REVIEW

**Comment by MohammadJavadD** [comment]
We also have added the code to download and prepare both datasets (TUAB and NMT) and then training and fine-tuning steps. the code works with the newest TUAB dataset and Braindecode. Please let me know if you have any questions. [APD_EEG repo](https://github.com/MohammadJavadD/APD_EEG).

**Comment by bruAristimunha** [comment]
Yes, @MohammadJavadD, you read my mind. this is my block with this PR and the NMT dataset PR. I am thinking of something....

File: examples/advanced_training/transfer_learning.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+# EEG data and transfer the model to NMT [3] EEG dataset. We follow the approach of [1]
+
+random_state = 2024
+n_jobs = 1
+
+# ## Loading and preprocessing the dataset
+#
+# ### Load and save the raw recordings
+#
+# Here we assume you already load and preprocess the raw recordings for both TUAB and NMT datasetsand saved the file in `TUAB_path' and 'NMT_path' respectively. To read more see this notebook [here](https://braindecode.org/stable/auto_examples/applied_examples/plot_tuh_eeg_corpus.html)
```
Comment (bruAristimunha) [inline examples/advanced_training/transfer_learning.py:52]:
```suggestion
# Here, we assume you have already loaded and preprocessed the raw recordings for both TUAB and NMT datasets and saved the file in `TUAB_path` and `NMT_path` respectively. To read more, see how to load the Temple University Dataset in this braindecode [tutorial](https://braindecode.org/stable/auto_examples/applied_examples/plot_tuh_eeg_corpus.html)
```

File: examples/advanced_training/transfer_learning.py [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+TUAB_ds = load_concat_dataset(
+    TUAB_path,
+    preload=False,
+    target_name=["pathological", "age", "gender"],  # )
+    ids_to_load=range(20),
+)
+
+######################################################################
+# Target is being set to pathological
+# -------------------------------------
```
Comment (bruAristimunha) [inline examples/advanced_training/transfer_learning.py:75]: The concept of target needs to be clear at the begin of the tutorial.

File: examples/advanced_training/transfer_learning.py [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+    pd.DataFrame([d.description for d in TUAB_ds.datasets]), overwrite=True
+)
+
+# Splitting dataset into train, valid and test sets
+#
+# We split the recordings by subject into train, validation and
+# testing sets.
+#
+
+# split based on train split from dataset
```
Comment (bruAristimunha) [inline examples/advanced_training/transfer_learning.py:93]: Explain a little more about the true and false here, related with abnormal classification

File: examples/advanced_training/transfer_learning.py [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+train_set = TUAB_ds.split("train")["True"]
+test_set = TUAB_ds.split("train")["False"]
+
+
+######################################################################
+# Create the models
+# -------------------------------------
+# We can now create the deep learning model.
+# In this tutorial, we use DeepNet introduced in [4].
```
Comment (bruAristimunha) [inline examples/advanced_training/transfer_learning.py:102]: Can you discuss a little more about the choice of the parameters?

Comment (bruAristimunha) [reply examples/advanced_training/transfer_learning.py:102]: You can directly use your paper, and say something, as investigated in [1] ...

File: examples/advanced_training/transfer_learning.py [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+    ],
+    device="cuda" if cuda else "cpu",
+)
+
+clf.initialize()  # This is important!
+print("classifier initialized")
+print(
+    "Number of parameters = ",
+    sum(p.numel() for p in model.parameters() if p.requires_grad),
+)
```
Comment (bruAristimunha) [inline examples/advanced_training/transfer_learning.py:211]:
```suggestion
print(clf)
```
This is will print the parameters because of EEGClassifier

File: examples/advanced_training/transfer_learning.py [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+    device="cuda" if cuda else "cpu",
+    max_epochs=n_epochs,
+)
+
+clf.initialize()  # This is important!
+print("classifier initialized")
+print(
+    "Number of parameters = ",
+    sum(p.numel() for p in model.parameters() if p.requires_grad),
+)
```
Comment (bruAristimunha) [inline examples/advanced_training/transfer_learning.py:365]:
```suggestion
print(clf)
```

File: examples/advanced_training/transfer_learning.py [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+clf.fit(window_train_set, y=None)
+
+######################################################################
+# Conclusion
+# -------------------------------------
+#
+# In this example, we used transfer learning (TL) as a way to learn
+# representations from a large EEG data and transfer to a smaller dataset.
+# You can put one of the results from your paper here.
+
```
Comment (bruAristimunha) [inline examples/advanced_training/transfer_learning.py:378]: You can put more information about why this is important and things like this

**Review by bruAristimunha — DISMISSED** [review]
Hi @MohammadJavadD,

I made a rapid initial review. We need a little more text/concepts and some figures. Maybe put a static figure from your paper or my paper.

Can you work a little on the text that comments?
